lib/instrumentation/http-shared.js

```diff
@@ -74,19 +75,52 @@ function isRequestBlacklisted (agent, req) {
   return false
 }
 
+// NOTE: This will also stringify and parse URL instances
+// to a format which can be mixed into the options object.
+function ensureUrl (v) {
+  if (typeof v === 'string' || v instanceof url.URL) {
+    return url.parse(String(v))
+  } else {
+    return v
+  }
+}
+
 exports.traceOutgoingRequest = function (agent, moduleName) {
   var spanType = 'ext.' + moduleName + '.http'
   var ins = agent._instrumentation
 
   return function (orig) {
-    return function () {
+    return function (...args) {
       var span = agent.buildSpan()
       var id = span && span.transaction.id
 
       agent.logger.debug('intercepted call to %s.request %o', moduleName, { id: id })
 
-      var req = orig.apply(this, arguments)
+      var options = {}
+      var newArgs = [ options ]
+      for (let arg of args) {
+        if (typeof arg === 'function') {
+          newArgs.push(arg)
```

Comment:
hopefully there will not be more than one callback 😬😄

Reply:
If there _was_, it'd find them just fine. I'm a bit wary of the argument folding thing for everything else though.

I need to do the folding to be able add the header to the inputs. In 10.9+, it supports `(urlString, options, ...)`, but less than that and it will only allow `(urlString, ...)` or `(options, ...)`, so I need to ensure the URL is converted to an object. There's also a problem with the `URL` type being not directly compatible with the `options` structure, so it needs to be stringified and converted back to an object with `url.parse(...)`, to ensure it has a valid structure. 🤔

Reply:
Yeah, it's probably fine - I didn't mean that I wanted you to change anything. I was just wondering...